### PR TITLE
Features/bch/registries

### PR DESCRIPTION
**Pull request by blaisefonseca**

### CODE REVIEW

File: app/views/reports/bch/registries/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+define(['app', 'lodash', 'css!/app/css/registry.css', 'services/search-service', 'js/common', 'services/solr',
+		'./index'], function(app, _) { 'use strict';
+
+		app.controller("registryReportController", ['$scope', 'searchService', 'toastr', '$log', 'solr',
+			function ($scope, searchService, toastr, $log, solr) {
```
Comment (sbilodeau) [inline app/views/reports/bch/registries/index.js:5]: it's possible to make the controller anonymous if loaed by lazyloader by returning the injection array instead. then you can remove the `ng-controller` from the `index.html` 

```
return ['$scope', 'searchService', 'toastr', '$log', 'solr',  function ($scope, searchService, toastr, $log, solr) {
...
```

control is instanciated here. https://github.com/scbd/absch.cbd.int/blob/master/app/routes/common-routes.js#L270

File: app/views/reports/bch/registries/view-lmo-registry.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			$scope.readyForExport = true;
+			require(['tableexport'], function () {
+				$element.find('#forExport').tableExport({
+					formats: ['xlsx'],
+					filename: 'LMO-registry',
+				});
+				$element.find('.xlsx').click();
+				$timeout(function () {
+					$scope.readyForExport = false;
+				}, 200)
```
Comment (sbilodeau) [inline app/views/reports/bch/registries/view-lmo-registry.js:46]: Avoid usage of timeout

```suggestion
				$scope.$applyAsync(function () {
					$scope.readyForExport = false;
				})
```

**Review by sbilodeau — COMMENTED** [review]
Try avoid using timeout as it is not predictable and may look liek a workAround for another problem. use $apply or $applyAsync to execute mutation into angular digest cycle. 

Unless mandatory it's preferable to use anonymous controller for views.

**Review by sbilodeau — APPROVED** [review]

File: app/views/reports/bch/registries/view-gene-registry.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+										<th class="display-18">Name</th>
+										<th class="display-25 hidden-xs">Trait(s)</th>
+										<th class="display-15 hidden-xs">Donor organism</th>
+										<th class="display-25 hidden-xs">Function</th>
+									</tr>
+								</thead>
+								<tbody>
+									<tr ng-repeat="row in ::geneRecords">
+										<td class="tableexport-string">
+											<a rel="noopener" translation-url target="_blank" ng-if="::row.url_ss.length" rel="noopener" ng-href="{{::row.url_ss[0]}}">
```
Comment (randyhoulahan) [inline app/views/reports/bch/registries/view-gene-registry.html:52]: rel="noopener"  x2 - not fatal

File: app/views/reports/bch/registries/view-lmo-registry.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+										<th class="display-18">Record ID</th>
+										<th class="display-25">Identity & transformation event</th>
+										<th class="display-17 hidden-xs">Organism</th>
+										<th class="display-40 hidden-xs">Description</th>
+									</tr>
+								</thead>
+								<tbody>
+									<tr ng-repeat="row in ::lmoRecords">
+										<td class="tableexport-string">
+											<a rel="noopener" translation-url target="_blank" ng-if="::row.url_ss.length" rel="noopener" ng-href="{{::row.url_ss[0]}}">
```
Comment (randyhoulahan) [inline app/views/reports/bch/registries/view-lmo-registry.html:58]: rel="noopener" x2

**Review by vincee — APPROVED** [review]

File: app/views/reports/bch/registries/view-orga-registry.html [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+										<th class="display-18">Record ID</th>
+										<th class="display-20">Taxonomic status</th>
+										<th class="display-17 hidden-xs">Common name</th>
+										<th class="display-45 hidden-xs">Description</th>
+									</tr>
+								</thead>
+								<tbody>
+									<tr ng-repeat="row in ::orgaRecords">
+										<td class="tableexport-string">
+											<a rel="noopener" translation-url target="_blank" ng-if="::row.url_ss.length" rel="noopener" ng-href="{{::row.url_ss[0]}}">
```
Comment (randyhoulahan) [inline app/views/reports/bch/registries/view-orga-registry.html:53]: need escape?

**Pull request merged**